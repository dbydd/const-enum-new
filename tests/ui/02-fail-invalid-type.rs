#![feature(const_trait_impl)]

use const_enum::ConstEnum;

#[derive(ConstEnum)]
struct TestStruct {}

#[derive(ConstEnum)]
union TestUnion {}

pub fn main() {}
