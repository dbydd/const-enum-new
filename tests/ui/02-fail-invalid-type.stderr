error: proc-macro derive panicked
 --> tests/ui/02-fail-invalid-type.rs:5:10
  |
5 | #[derive(ConstEnum)]
  |          ^^^^^^^^^
  |
  = help: message: unexpected struct, const-enum only supports enums

error: proc-macro derive panicked
 --> tests/ui/02-fail-invalid-type.rs:8:10
  |
8 | #[derive(ConstEnum)]
  |          ^^^^^^^^^
  |
  = help: message: unexpected union, const-enum only supports enums

error: unions cannot have zero fields
 --> tests/ui/02-fail-invalid-type.rs:9:1
  |
9 | union TestUnion {}
  | ^^^^^^^^^^^^^^^^^^
